ui-spacetimechart/src/components/workSchedules/WorkSchedules.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+import Region from './Region';
+import { type WorkSchedule } from '../../types';
+
+type WorkSchedulesProps = {
+  workSchedules: WorkSchedule[];
+  pattern: string;
+};
+/**
+ * Display the workschedule projection on the Space time chart.
+ * a workschedule can contains several track occupied during a period of this
+ * each is represented by a rectangle region
+ */
+function WorkSchedules({ workSchedules, pattern }: WorkSchedulesProps) {
+  return workSchedules.flatMap((ws) =>
+    ws.intersections.map(([spaceStart, spaceEnd]) => (
+      <Region
```

Comment:
Is it fine to create one separate component per rect? Wouldn't it be better to pass a list of rects to a component responsible for drawing the pattern?

That way, we wouldn't end with a big list of components in the DOM and we wouldn't load the pattern image each time we draw an individual rect.

Reply:
I think this is already pretty optimised. Region doesn’t create anything in the DOM (return `null`). It just calls the canvas context apis.

Reply:
Indeed. Still, many `Image` objects are created, each resulting in an HTTP request (potentially cached) and SVG parsing etc.

Reply:
Ah yes, good point. I agree with @emersion

Reply:
I moved the image loading to the parent

Reply:
I moved the image loading to the parent @emersion @clarani 
what do you think?

Reply:
I think the performance concerns are addressed. I still believe a layer painting a region is a better design than a layer painting a single rect (see https://github.com/OpenRailAssociation/osrd-ui/pull/523#discussion_r1801149689 for another reason why) but since it's not public API I'm fine with keeping it as-is.

Reply:
Merging for now, we can rework this when need arises